### ISSUE TITLE
Athena Assembly olc error

### ISSUE THREAD

**Issue by knguyenjcvi** [abishara/athena_meta#10]
Hello,

I have a metagenomic dataset that I would like to assembly using athena. However I am running into this issue, which I have attached. Can you please help me identify a solution?

Thanks in Advance,
Kevin Nguyen
[athena_olc_error.txt](https://github.com/abishara/athena_meta/files/2059182/athena_olc_error.txt)

**Comment by aslomko** [comment]
@abishara I am getting the same error as @knguyenjcvi .  Was there every any help regarding this error?

**Comment by abishara** [comment]
Can you upload your log file?  I can take a look. 
alex

**Comment by aslomko** [comment]
> Can you upload your log file? I can take a look.
> alex

The relevant portion of the error on assemble_olc

##Removed data

**Comment by abishara** [comment]
Hmm, this looks like something may be a bit funky with your bwa/samtools binary/install.  Can you successfully run bwa independently of athena with it finishing successfully?

The command that is failing is:

% bwa mem -t 4 /home/slomka/CDiff/spadesoutCdiff/contigs.fasta /home/slomka/CDiff/results/olc/pre-flye-input-contigs.fa | samtools view -bS - | samtools sort -o /home/slomka/CDiff/results/olc/align-inputs.bam -

How about trying it on the first 8 lines of pre-flye-input-contigs.fa, which is just the first two reads to see if we can narrow it down if it isn't your bwa/samtools install?  If this fails, can you upload those two reads as well as the reference /home/slomka/CDiff/spadesoutCdiff/contigs.fasta 

Thanks!

**Comment by aslomko** [comment]
so it looks like the pre-flye-input-contigs.fa is actually empty on my system. 0kb empty file.  I'm on flye 2.3.5

**Comment by abishara** [comment]
Were you able to successfully run the small example dataset on the github:

https://storage.googleapis.com/gbsc-gcp-lab-bhatt-public/readclouds-l-gasseri-example.tar.gz

with your install?  Also can you copy the result of the following, which should be executed relative to your run directory:

% ls -lhS ./results/bin.*/*fa | head

Let's see if any of the subassemblies actually went through.

**Comment by aslomko** [comment]
The result is

-rw-r--r-- 1 slomka slomka 0 Sep 24 11:47 ./results/bin.0/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.100/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.101/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.102/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.103/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.104/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.105/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.106/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.107/local-asm-merged.fa
-rw-r--r-- 1 slomka slomka 0 Sep 24 11:50 ./results/bin.108/local-asm-merged.fa

I also tried running the example dataset and it failed as well.  I never mentioned that I was running on lsf setup.  When running on a regular server (i tried with the example dataset after it gets passed the part where on the lsf it fails.

**Comment by abishara** [comment]
Can you try to get the example dataset to go through successfully first using multiprocessing mode (and local mode if that fails) so we can determine that the environment is setup so that it can run properly?  There is also a docker image referenced in the github that can be used to help get the dependencies properly set up.

Once this goes through, then perhaps we can try the example dataset with the lsf setup.  Although I will mention that I am planning to phase out support soon for running through queueing submission systems in favor of running it on 32+ core machines, which works well and is straightforward.

Thanks!

**Comment by aslomko** [comment]
> Can you try to get the example dataset to go through successfully first using multiprocessing mode (and local mode if that fails) so we can determine that the environment is setup so that it can run properly? There is also a docker image referenced in the github that can be used to help get the dependencies properly set up.
> 
> Once this goes through, then perhaps we can try the example dataset with the lsf setup. Although I will mention that I am planning to phase out support soon for running through queueing submission systems in favor of running it on 32+ core machines, which works well and is straightforward.
> 
> Thanks!

I got the example running multiprocessing mode.  I actually did not change anything in the config file (i already had cluster setting multiprocessing but was then sending the job to a lsf.  I wasn't quite sure how to set up the json file for an actual lsf cluster

**Comment by aslomko** [comment]
I also think I've narrowed it down to the step before olc generation for my data.  Like i said before somehow the pre-flye-input-contigs is empty and I don't know why.  Looking at the logs i get
 seed NODE_138_length_1412_cov_4351.597004 contig does not have high enough coverage
for all assemble bins

**Comment by abishara** [comment]
Great, glad you were able to get it running. That file is likely empty because none of the subassemblies actually went through.  It's possible that the barcodes are not attached as expected to either the input *fastq or *bam and that it still went through this far in process.  I am working on providing better error checking for inputs to catch that earlier in the process.

For now, if you are willing to send me an email with a link to the inputs, I can have a look.  This would be the easiest way for me to figure this out.

**Comment by aslomko** [comment]
I actually figured out my issue,  it was with my .fq file, it did not have a tab before the BC tag so there were actually no barcodes being read by flye.  After that it fixed my issue.

**Comment by abishara** [comment]
Hi,

Thanks for the update and I had a feeling this might be the case.  I'm hoping to make a release that will check for these and provide an error/warning if the inputs are not tagged with the barcodes as expected. 

If you haven't figured out already, I advise you rerun everything from the beginning (remove your scratch/working directory) so that the barcodes are seen thru the entire process.

alex

**Comment by nick-youngblut** [comment]
I was getting the same error with athena_meta v1.1 due to the "pre-flye-input-contigs.fa" file being empty. My fastq read headers looked like the following:

```
@NS500418:354:H27G3BGXY:3:12612:25572:11380 RG:Z:readgroup1 BC:Z:GCCAATTCAAGTTT-1
```

...which is shown in the athena_meta readme. I removed the "RG:Z:readgroup1" section from the read headers and that fixed the problem.

**Comment by abishara** [comment]
It turns out that the additional tags actually have to be tab-delimited to work with 'bwa mem -C' otherwise the barcodes are not properly detected in the *bam.  Commit 02f26e74d1fc64fc specifies this additional restriction.  Better error checking for the input *fastq and *bam have also been included in commit 11dbe60f0d1383.
